tests/5.2/declare_target/test_declare_target_enter_device_type_nohost.c

```diff
@@ -0,0 +1,92 @@
+//===---- test_declare_target_enter_device_type_nohost.c  ---------------------------===//
+// 
+// OpenMP API Version 5.0 
+//
+// The declare target directive specifies that variables, functions(C,C++ and Fortran),
+// and subroutines (Fortran) are mapped to a device. If a device_type
+// clause is present on the contained declare target directive, then its argument 
+// determines which versions are made available. If device_type(nohost) is present 
+// only a device version of the procedure is made available. 
+// The device version of the function has to be specified via declare variant to support
+// fallback execution.
+// 
+//===-------------------------------------------------------------------------------===//
+
+#include <omp.h>
+#include <stdio.h>
+#include <assert.h>
+#include <stdlib.h>
+#include "ompvv.h"
+
+#define N 10
+int errors = 0;
+int a[N], b[N], c[N];  
+int dev=-9, i = 0;
+
+void target_function();
+#pragma omp declare target enter(target_function) device_type(nohost)
+#pragma omp declare target enter(a,b,c,i,dev)
+
+#pragma omp declare variant(target_function) match(device={kind(nohost)})
+void update() {
+  for (i = 0; i < N; i++) {
+    a[i] += 5;
+    b[i] += 5;
+    c[i] += 5;
+  }
+}
+
+void target_function(){
+  for (i = 0; i < N; i++) {
+    a[i] += 1;
+    b[i] += 2;
+    c[i] += 3;
+  }
+} 
+
+int test_declare_target_enter_device_type_nohost() { 
+
+  update();
+
+  #pragma omp target update to(a,b,c)
+
+  #pragma omp target  
+  {
+    update();
```

Comment:
As in the test:  tests/5.2/declare_target/test_declare_target_enter_device_type_nohost1.c
Would be useful to comment
//Will call target_function() on device OR update() in case of fall-back